### ISSUE TITLE
stochastic encoding?

### ISSUE THREAD

**Issue by MustafaMustafa** [mkusner/grammarVAE#9]
Hi Matt,

Thank you very much for posting the code of your very nice work.

You mention in the paper appendix, in the context of calculating the reconstruction accuracy, that you encode each of the 5000 test molecules 10 time "as encoding and decoding are stochastic". Looking at your the code here, I can see that the decoding is stochastic, but the encoding looks deterministic to me: https://github.com/mkusner/grammarVAE/blob/master/molecule_vae.py#L71-L84
Where does the stochasticity of the encoding come in?

I changed the keras imports to use `tf.keras` and planning to train it using cuda 9.0 and TF 1.8.0-dev. I hope I will be able to reproduce your results. Did you obtain the `./pretrained` weights using the hyperparameters in this code repo? For example, did you use default Adam learning rate?

**Comment by mkusner** [comment]
Hey Mustafa! Thanks for your interest in the code!

To your first question: if you change line 84 to: self
.vae.encoder.predict(one_hot)
then you get the stochastic encoder. The molecule_vae.py code was mainly
created so one could quickly start using the code and was not used for
evaluation in the paper. But this change will allow you to compute the
metric you describe.

Yes the code should produce the pretrained output, but please see the
closed issue #6 for additional tips about this!

Matt





On Apr 17, 2018 12:32 AM, "Mustafa Mustafa" <notifications@github.com>
wrote:

Hi Matt,

Thank you very much for posting the code to your very nice work.

You mention in the paper appendix, in the context of calculating the
reconstruction accuracy, that you encode each of the 5000 test molecules 10
time "as encoding and decoding are stochastic". Looking at your the code
here, I can see that the decoding is stochastic, but the encoding looks
deterministic to me:
https://github.com/mkusner/grammarVAE/blob/master/molecule_vae.py#L71-L84
Where does the stochasticity of the encoding come in?

I changed the keras imports to use tf.keras and planning to train it using
cuda 9.0 and TF 1.8.0-dev. I hope I will be able to reproduce your results.
Did you obtain the './pretrainedweights using the hyperparameters in this
code repo? For example, did you really use default Adam learning rate for
the results in the paper and./pretraied` weights?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mkusner/grammarVAE/issues/9>, or mute the thread

**Comment by MustafaMustafa** [comment]
Thank for the prompt response Matt. I tested quickly and I can get ~54% reconstruction efficiency with ~~default~~ modified parameters and more recent TF, Keras, Cuda, Cudann codes (1.8.0, tf.keras, 9.0, 7.0.5 respectively).
